flow/layers/child_scene_layer.cc

```diff
@@ -19,10 +19,25 @@ ChildSceneLayer::ChildSceneLayer(zx_koid_t layer_id,
 
 void ChildSceneLayer::Preroll(PrerollContext* context, const SkMatrix& matrix) {
   set_needs_system_composite(true);
+
+  // An alpha "hole punch" is required if the frame behind us is not opaque.
+  if (!context->is_opaque) {
+    set_paint_bounds(
+        SkRect::MakeXYWH(offset_.fX, offset_.fY, size_.fWidth, size_.fHeight));
+  }
 }
 
 void ChildSceneLayer::Paint(PaintContext& context) const {
```

Comment:
Can we have a unit test or an engine golden test (like [performance_overlay_layer_unittest.cc](https://github.com/flutter/engine/blob/master/flow/layers/performance_overlay_layer_unittests.cc)) to verify this behavior?